unable to download my music playlist

### Steps to reproduce

1. Import playlist https://youtube.com/playlist?list=PL_9f-TXRu6eab7pgBfpdiDuN9zr5SR_Wy&si=Jo4sAmfnBXuyKaeK
2. Download videos
3.then I get this error.
 java.lang.RuntimeException: Unable to start service com.github.libretube.services.PlaylistDownloadEnqueueService@b6ab85b with null: java.lang.NullPointerException
	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5246)
	at android.app.ActivityThread.-$$Nest$mhandleServiceArgs(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2546)
	at android.os.Handler.dispatchMessage(Handler.java:109)
	at android.os.Looper.loopOnce(Looper.java:232)
	at android.os.Looper.loop(Looper.java:317)
	at android.app.ActivityThread.main(ActivityThread.java:8787)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:591)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:871)
Caused by: java.lang.NullPointerException
	at com.github.libretube.services.PlaylistDownloadEnqueueService.onStartCommand(Unknown Source:13)
	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5228)
	... 9 more
    

### Expected behavior

Download my playlist

### Actual behavior

Crashes when downloading

### LibreTube version

0.27.1

### Android version

Android 15

### Other details




### Acknowledgements

- [x] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [x] I have written a short but informative title.
- [x] I will fill out all of the requested information in this form.

--- a/app/src/main/java/com/github/libretube/services/PlaylistDownloadEnqueueService.kt
+++ b/app/src/main/java/com/github/libretube/services/PlaylistDownloadEnqueueService.kt
@@ -23,6 +23,7 @@ import com.github.libretube.extensions.toID
 import com.github.libretube.extensions.toastFromMainDispatcher
 import com.github.libretube.helpers.DownloadHelper
 import com.github.libretube.parcelable.DownloadData
+import com.github.libretube.util.TextUtils
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
 
@@ -142,9 +143,16 @@ class PlaylistDownloadEnqueueService : LifecycleService() {
             val videoStream = getStream(videoInfo.videoStreams, maxVideoQuality)
             val audioStream = getStream(videoInfo.audioStreams, maxAudioQuality)
 
+            // remove all UNIX reserved characters from the title in order to generate
+            // a valid filename
+            var fileName = videoInfo.title
+            TextUtils.RESERVED_CHARS.forEach {
+                fileName = fileName.replace(it, '_')
+            }
+
             val downloadData = DownloadData(
                 videoId = stream.url!!.toID(),
-                fileName = videoInfo.title,
+                fileName = fileName,
                 videoFormat = videoStream?.format,
                 videoQuality = videoStream?.quality,
                 audioFormat = audioStream?.format,

